frontend/src/metabase/query_builder/components/NativeQueryEditor/__mocks__/NativeQueryEditor.jsx

```diff
@@ -1,14 +1,46 @@
 /* eslint-disable react/prop-types */
 import SyncedParametersList from "metabase/parameters/components/SyncedParametersList/SyncedParametersList";
 import { ACE_ELEMENT_ID } from "metabase/query_builder/components/NativeQueryEditor/constants";
+import DataSourceSelectors from "metabase/query_builder/components/NativeQueryEditor/DataSourceSelectors";
 
-const MockNativeQueryEditor = ({ query, setParameterValue, ...props }) => {
+const MockNativeQueryEditor = ({
+  canChangeDatabase = true,
+  editorContext = "question",
+  isNativeEditorOpen,
+  query,
+  readOnly,
+  setDatasetQuery,
+  setParameterValue,
+}) => {
   const onChange = evt => {
-    props.setDatasetQuery(query.setQueryText(evt.target.value));
+    setDatasetQuery(query.setQueryText(evt.target.value));
+  };
+
+  const onDatabaseIdChange = databaseId => {
+    if (query.databaseId() !== databaseId) {
+      setDatasetQuery(query.setDatabaseId(databaseId).setDefaultCollection());
+    }
+  };
+
+  const onTableIdChange = tableId => {
+    const table = query.metadata().table(tableId);
+    if (table && table.name !== query.collection()) {
+      setDatasetQuery(query.setCollectionName(table.name));
+    }
   };
 
   return (
     <div data-testid="mock-native-query-editor" id={ACE_ELEMENT_ID}>
+      {canChangeDatabase && (
+        <DataSourceSelectors
```

Comment:
When rendering `DataSelector`, [`componentDidUpdate`](https://github.com/metabase/metabase/blob/78272851594b37499837d23f0de63f6afdfe24b6/frontend/src/metabase/query_builder/components/DataSelector/DataSelector.jsx#L424-L484) will [call `switchToStep`](https://github.com/metabase/metabase/blob/78272851594b37499837d23f0de63f6afdfe24b6/frontend/src/metabase/query_builder/components/DataSelector/DataSelector.jsx#L471), which will [call `skipSteps`](https://github.com/metabase/metabase/blob/78272851594b37499837d23f0de63f6afdfe24b6/frontend/src/metabase/query_builder/components/DataSelector/DataSelector.jsx#L746), which will [call `onChangeDatabase`](https://github.com/metabase/metabase/blob/78272851594b37499837d23f0de63f6afdfe24b6/frontend/src/metabase/query_builder/components/DataSelector/DataSelector.jsx#L554) and set `database` in the new `Question`'s `query`.

Without it, question would be in an _uninitialized_, `readOnly` state and it would not matter for the unsaved changes warning.

That's why I had to add `DataSourceSelectors` to this mock. It's the same as in the real [`NativeQueryEditor`](https://github.com/metabase/metabase/blob/1efa6c6f000cb4bd38065563176c35ea0027c865/frontend/src/metabase/query_builder/components/NativeQueryEditor/NativeQueryEditor.tsx#L770-L779).